presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -933,32 +1105,76 @@ public void testFilterContainmentWithMismatchStringLength()
-
-        Query originalViewQuery = (Query) SQL_PARSER.createStatement(originalViewSql, PARSING_OPTIONS);
-        Query baseQuery = (Query) SQL_PARSER.createStatement(baseQuerySql, PARSING_OPTIONS);
-        Query expectedViewQuery = (Query) SQL_PARSER.createStatement(expectedViewSql, PARSING_OPTIONS);
-
         transaction(transactionManager, accessControl)
                 .singleStatement()
                 .readUncommitted()
                 .readOnly()
                 .execute(CLIENT_SESSION, session -> {
-                    Query optimizedBaseToViewQuery = (Query) new MaterializedViewQueryOptimizer(metadata, session, SQL_PARSER, accessControl, domainTranslator, viewTable, originalViewQuery).rewrite(baseQuery);
+                    Query baseQuery = (Query) SQL_PARSER.createStatement(baseQuerySql, createParsingOptions(session));
+                    Query expectedViewQuery = (Query) SQL_PARSER.createStatement(expectedQueryAfterOptimization, createParsingOptions(session));
+                    Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViewMap = new HashMap<>();
+
+                    int i = 1;
+                    for (String existingViewQuery : existingViewQueries) {
+                        QualifiedObjectName viewTableName = buildQualifiedNameFromViewQuery(session, existingViewQuery);
+                        baseTableToMaterializedViewMap.computeIfAbsent(viewTableName, (x) -> new ArrayList<>());
+                        baseTableToMaterializedViewMap.get(viewTableName).add(createStubConnectorMaterializedViewDefinition(existingViewQuery, i++));
+                    }
+
+                    Query optimizedBaseToViewQuery = new MaterializedViewQueryOptimizer(metadata, session, SQL_PARSER, accessControl, domainTranslator, baseTableToMaterializedViewMap).rewrite(baseQuery);
                     assertEquals(optimizedBaseToViewQuery, expectedViewQuery);
                 });
     }
+
+    private QualifiedObjectName buildQualifiedNameFromViewQuery(Session session, String existingViewQuery)
+    {
+        try {
+            Query viewQuery = (Query) SQL_PARSER.createStatement(existingViewQuery, createParsingOptions(session));
+            QuerySpecification querySpecification = (QuerySpecification) viewQuery.getQueryBody();
+            if (querySpecification.getFrom().get() instanceof Table) {
+                Table table = (Table) querySpecification.getFrom().get();
+                return createQualifiedObjectName(session, table, table.getName());
+            }
+            else if (querySpecification.getFrom().get() instanceof AliasedRelation) {
+                AliasedRelation aliasedRelation = (AliasedRelation) querySpecification.getFrom().get();
+                Table table = (Table) aliasedRelation.getRelation();
+                return createQualifiedObjectName(session, table, table.getName());
+            }
+            else {
```

Comment:
else not needed here and so on

Reply:
Regarding unnecessary `else`- I tend to treat it on a case by case basis with readability in mind. Given we don't have a checkstyle rule, I'd lean towards keeping these in. WDYT?

Reply:
Lets remove the else. Less code and less indentation is generally better.